### DISCUSSION TITLE
How to delete edit history of github issue using API

### DISCUSSION THREAD

**Discussion by youngbupark** [community/community#63882] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hi All, 

I am looking for the way to delete edit history of github issue using API. I found the way to fetch the edit history of github issues using graphql api, but I was unable to find the mutation graphql api and rest api to delete edit history of github issues. Is there a public API to delete edit history?

```graphql
query ($owner: String!, $name: String!) {
  repository(name: $name, owner: $owner) {
    issues(first: 10) {
      nodes {
        userContentEdits(first: 10) {
          nodes {
            diff
            id
          }
        }
        comments(first: 10) {
          nodes {
            userContentEdits(first: 10) {
              nodes {
                diff
                id
                createdAt
              }
            }
            id
          }
        }
      }
    }
  }
}
```

**Comment by nirav-gajera** [comment, marked as answer]
maybe ,  GitHub does not allow users to delete edit history for security reasons. This is because the edit history can be used to track changes to issues and comments, which can be helpful for debugging and tracking the progress of a project.

If you need to delete the edit history of a GitHub issue, you can contact GitHub support and request that they delete it for you. However, they may not be able to do so, depending on the reason why you need to delete the edit history

**Reply by youngbupark** [reply]
Thanks! I also got the response from GitHub support. There is no public API for deleting edit-history.
